p2p/client.go

```diff
@@ -431,6 +541,90 @@ func (c *Client) bootstrapLoop(ctx context.Context) {
 	}
 }
 
+// latest seen height to be retrieved using block-sync
+func (c *Client) setLatestSeenHeight(height uint64) {
+	c.latestSeenHeight = max(height, c.latestSeenHeight)
+}
+
+// retrieveBlockSyncLoop checks if there is any missing block to be retrieved and requests it on demand
```

Comment:
definition of missing block? how does it knows a block is missing?

Reply:
block missing is a block not received from p2p, that is between the latest received height and the latest block applied